index.js

```diff
@@ -403,17 +419,22 @@ function keysEqual(leftHandOperand, rightHandOperand, keys, options) {
 
 /*!
  * Recursively check the equality of two Objects. Once basic sameness has been established it will defer to `deepEqual`
- * for each enumerable key in the object.
+ * for each enumerable key in the object. For `Error` objects, also compare `name` and `message` keys, regardless of
+ * enumerability.
  *
  * @param {Mixed} leftHandOperand
  * @param {Mixed} rightHandOperand
+ * @param {String} type of leftHandOperand
  * @param {Object} [options] (Optional)
  * @return {Boolean} result
  */
-
-function objectEqual(leftHandOperand, rightHandOperand, options) {
+function objectEqual(leftHandOperand, rightHandOperand, leftHandType, options) {
   var leftHandKeys = getEnumerableKeys(leftHandOperand);
   var rightHandKeys = getEnumerableKeys(rightHandOperand);
+  if (leftHandType === 'Error') {
+    addExtraErrorKeys(leftHandKeys);
+    addExtraErrorKeys(rightHandKeys);
```

Comment:
I think it'd be simpler just to do:

```js
leftHandkeys.push('name');
leftHandKeys.push('message');
rightHandKeys.push('name');
rightHandKeys.push('message');
```

Calling out to a function and checking if those keys are already in the index is "probably" (my hand-wavy intuition) more costly than just checking the property for equality twice over. Certainly the code would be cleaner though

Reply:
I like the simpler form aesthetically, but the problem is that if one `Error` object is assigned a custom `name` property (thus shadowing its prototype's `name`) and the other one isn't, then `name` gets added twice to the list of keys for the object that has the custom `name` property (once from `getEnumerableKeys`, and once from the manual `push`), whereas the other object only gets it added once from the manual `push`. The comparison then fails due to a difference in length of the two key arrays.